Move accent to protocol

ConfusedVorlon opened the pull request:
This works for buttons, though not for toggles.
I can't see why toggles fail as it's using the same approach

chkn commented on Readme.md:
````diff
@@ -65,6 +65,26 @@ Button("Text Button", action: {})
+            .accentColor(Color.green)
+```
+
+In MacOS the accentColor modifier is not available, so you can use the materialColor environment variable
````
```suggestion
In MacOS the accentColor modifier is not available, so you can use the materialAccent environment variable
```

chkn commented on Sources/MaterialUI/Button.swift:
```diff
@@ -29,14 +29,15 @@ struct MaterialButtonContent: ViewModifier {
 public struct ContainedButtonStyle: PrimitiveButtonStyle {
 	private struct LabelModifier: ViewModifier {
 		@Environment(\.isEnabled) var isEnabled: Bool
+        @Environment(\.materialAccent) var accentColor: Color
```
use tabs instead of spaces
```suggestion
		@Environment(\.materialAccent) var accentColor: Color
```

chkn commented on Sources/MaterialUI/Button.swift:
```diff
@@ -58,16 +59,17 @@ public struct ContainedButtonStyle: PrimitiveButtonStyle {
 public struct OutlinedButtonStyle: PrimitiveButtonStyle {
 	private struct LabelModifier: ViewModifier {
 		@Environment(\.isEnabled) var isEnabled: Bool
+        @Environment(\.materialAccent) var accentColor: Color
```
```suggestion
		@Environment(\.materialAccent) var accentColor: Color
```

chkn commented on Sources/MaterialUI/Button.swift:
```diff
@@ -88,13 +90,14 @@ public struct OutlinedButtonStyle: PrimitiveButtonStyle {
 public struct TextButtonStyle: PrimitiveButtonStyle {
 	private struct LabelModifier: ViewModifier {
 		@Environment(\.isEnabled) var isEnabled: Bool
+        @Environment(\.materialAccent) var accentColor: Color
```
```suggestion
		@Environment(\.materialAccent) var accentColor: Color
```

chkn commented on Sources/MaterialUI/MaterialColor.swift:
```diff
@@ -0,0 +1,37 @@
+//
```
To be consistent with the rest of the code, this file should use tabs instead of spaces

chkn commented on Sources/MaterialUI/MaterialColor.swift:
```diff
@@ -0,0 +1,37 @@
+    var materialAccent: Color {
+        get {
+            let color = self[MaterialAccentColorKey.self]
+            print("color get: \(color)")
```
Can we remove these print statements?

ConfusedVorlon replied:
sorry - forgotten debug statements

chkn commented on Sources/MaterialUI/MaterialColor.swift:
```diff
@@ -0,0 +1,37 @@
+            return color
+        }
+        set {
+            print("color set: \(newValue)")
```
Can we remove these print statements?

chkn commented on Sources/MaterialUI/Toggle.swift:
```diff
@@ -9,7 +9,9 @@
-	// https://github.com/flutter/flutter/blob/ee032f67c734e607d8ea5c870ba744daf4bf56e7/packages/flutter/lib/src/material/switch.dart#L19
+	@Environment(\.materialAccent) var accentColor: Color
+    
+    // https://github.com/flutter/flutter/blob/ee032f67c734e607d8ea5c870ba744daf4bf56e7/packages/flutter/lib/src/material/switch.dart#L19
```
spaces -> tabs again
```suggestion
	// https://github.com/flutter/flutter/blob/ee032f67c734e607d8ea5c870ba744daf4bf56e7/packages/flutter/lib/src/material/switch.dart#L19
```

chkn commented on Sources/MaterialUI/Toggle.swift:
```diff
@@ -9,7 +9,9 @@
 
 public struct MaterialSwitchToggleStyle: ToggleStyle {
-	// https://github.com/flutter/flutter/blob/ee032f67c734e607d8ea5c870ba744daf4bf56e7/packages/flutter/lib/src/material/switch.dart#L19
+	@Environment(\.materialAccent) var accentColor: Color
```
So `@Environment` doesn't seem to work inside of styles. See the note for Button: https://github.com/chkn/MaterialUI/blob/master/Sources/MaterialUI/Button.swift#L11

We should be able to just define a `ViewModifier` for this one like in the others

ConfusedVorlon replied:
moved switch into a view

chkn reviewed (commented):
This is great, thanks so much! Just a couple things: for consistency, we should use tabs instead of spaces. Also see my note inline about why it doesn't work for toggles.

Finally, just a small matter of business.. I just realized that there was no license on this repo. I've just added an MIT License and want to confirm that you're OK with contributing under those terms.

ConfusedVorlon committed:
Update Readme.md

Co-authored-by: Alex Corrado <alexander.corrado@gmail.com>

ConfusedVorlon committed:
use tabs

Co-authored-by: Alex Corrado <alexander.corrado@gmail.com>

ConfusedVorlon committed:
use tabs

Co-authored-by: Alex Corrado <alexander.corrado@gmail.com>

ConfusedVorlon committed:
use tabs

Co-authored-by: Alex Corrado <alexander.corrado@gmail.com>

ConfusedVorlon committed:
remove print statements

ConfusedVorlon committed:
spaces->tabs

Co-authored-by: Alex Corrado <alexander.corrado@gmail.com>

ConfusedVorlon committed:
Move the switch to a separate view

so it can use @Environment

ConfusedVorlon committed:
Merge branch 'feature/environment_accent' of https://github.com/ConfusedVorlon/MaterialUI into feature/environment_accent

ConfusedVorlon committed:
re-align

ConfusedVorlon commented:
hopefully good to go

ConfusedVorlon commented:
and - yes, I'm happy with MIT licence

chkn commented:
Merged! Thanks for your contribution

ConfusedVorlon committed:
Move accent to protocol

This works for buttons, though not for toggles.
I can't see why toggles fail as it's using the same approach
